statsd/statsd.go

```diff
@@ -288,6 +289,10 @@ func newWithWriter(w statsdWriter, o *Options, writerName string) (*Client, erro
 		Tags:          o.Tags,
 		telemetryTags: []string{clientTelemetryTag, clientVersionTelemetryTag, "client_transport:" + writerName},
 	}
+	if o.Aggregation {
+		c.agg = newAggregator(&c)
+		c.agg.start(o.AggregationFlushInterval)
```

Comment:
I think that's better to use the `go` keyword here instead of into `aggregator.start`, making it obvious _here_ that it'll be running concurrently.

Reply:
I don't agree actually. I think the aggregator should not expose it's internal behavior to the outside. The caller should not have to know how to start the aggregator (with a goroutine or not). From a user point of view, you create the `aggregator` and then it's `start` and `stop`.

The complexity should be encapsulate inside the class. WDYT ?

Reply:
Got your point!

> The caller should not have to know how to start the aggregator (with a goroutine or not).

That's where I differ because I think it is the user of the aggregator which is deciding to run it concurrently. (I'm not trying to convince you, just sharing my thoughts, I'm 100% fine with this solution 👍)

Reply:
When would you not want to run it concurrently ? I don't see the use case. By design the aggregator needs to run concurrently (so we can sample metrics on one side and aggregate on the other side).

Reply:
Yup that always needs concurrency; it is just that you don't know from here that starting the aggregator is non-blocking. Definitely not an "issue" just a standpoint, resolving this thread!